playbooks/common/openshift-master/config.yml

```diff
@@ -42,6 +42,12 @@
           public_hostname: "{{ openshift_public_hostname | default(None) }}"
           public_ip: "{{ openshift_public_ip | default(None) }}"
           deployment_type: "{{ openshift_deployment_type }}"
+          http_proxy: "{{ openshift_http_proxy | default(None) }}"
+          https_proxy: "{{ openshift_https_proxy | default(None) }}"
+          no_proxy: "{{ openshift_no_proxy | default(None) }}"
+          no_proxy_auto_include_hostnames: "{{ openshift_no_proxy_internal_hosts | default(False) }}"
+          no_proxy_internal_hostnames: "{{ hostvars | oo_select_keys(groups['oo_nodes_to_config'] | union(groups['oo_masters_to_config']) | union(groups['oo_etcd_to_config'] | oo_flatten() ))
+                                          | oo_collect('openshift.common.all_hostnames') | oo_flatten() | join(',')}}"
```

Comment:
We do have the g_all_hosts variable...  We could also add a oo_all_hosts to evaluate_groups for consistency.

 Also, I'm not sure that oo_flatten is needed after the union filters.